k4FWCore/include/k4FWCore/MetaDataHandle.h

```diff
@@ -73,7 +65,7 @@ MetaDataHandle<T>::MetaDataHandle(const Gaudi::DataHandle& handle, const std::st
 }
 
 //---------------------------------------------------------------------------
```

Comment:
Add docstring

Reply:
I have added a docstring that only mentions `std::optional`. Should it also mention that the return type depends on the version of podio? I am not entirely sure whether that does not confusing.

Reply:
Mentioning that this depends on podio version would be good, because people can see the std::optional or T in doxygen, but they can't see that this might change.

Reply:
Add the possibility to add a default value, and if no default value is given throw the exception.

Reply:
How about still having a version returning optional e.g.`get_optional`? I think it's pretty common to have throwing, default and optional getters for different cases. This way we don't mess-up with the return here but still provide a way to handle missing values when an exception is too much

Reply:
`get_optional` is done and the others are now using that and simply changing return values if necessary. Additionally, `get_optional` is now also the single place where all the podio version differences are handled and a `std::optional` is created on the fly with the heuristic check for empty values to return an empty optional.